src/app/models/ability.rb

```diff
@@ -0,0 +1,35 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    
+    user ||= User.new # nil guard; creats a new user object
+    if user.role?('admin')
+      can :manage, :all
+    elsif user.role?('teacher')
+      can :index, :dashboard
```

Comment:
Further down the road we will need to update this role so that teachers can `:manage` assignments, right? If I remember correctly, this current setup lets students and teachers view specific dashboards via the `:index` argument. If so, this lgtm